packages/keymanager-server/src/server.ts

```diff
@@ -0,0 +1,163 @@
+import fastify, {FastifyError, FastifyInstance} from "fastify";
+import fastifyCors from "fastify-cors";
+import bearerAuthPlugin from "fastify-bearer-auth";
+import querystring from "querystring";
+import {IncomingMessage} from "node:http";
+import {Api} from "@chainsafe/lodestar-api/keymanager";
+import {getRoutes} from "@chainsafe/lodestar-api/keymanager_server";
+import {registerRoutesGroup, RouteConfig} from "@chainsafe/lodestar-api/server";
+import {ErrorAborted, ILogger} from "@chainsafe/lodestar-utils";
+import {IChainForkConfig} from "@chainsafe/lodestar-config";
+import crypto from "node:crypto";
+import {unlink, writeFile} from "fs/promises";
+import {toHexString} from "@chainsafe/ssz";
+export {allNamespaces} from "@chainsafe/lodestar-api";
+
+export type RestApiOptions = {
+  host: string;
+  cors: string;
+  port: number;
+  auth: boolean;
+  tokenDir?: string;
+};
+
+export const restApiOptionsDefault: RestApiOptions = {
+  host: "127.0.0.1",
+  port: 9597,
+  cors: "*",
+  auth: true,
+};
+
+export interface IRestApiModules {
+  config: IChainForkConfig;
+  logger: ILogger;
+  api: Api;
+}
+
+const apiTokenFileName = "api-token.txt";
+
+export class KeymanagerServer {
+  private readonly opts: RestApiOptions;
+  private readonly server: FastifyInstance;
+  private readonly logger: ILogger;
+  private readonly activeRequests = new Set<IncomingMessage>();
+  private readonly apiTokenPath: string;
+  private readonly bearerToken: string;
+
+  constructor(optsArg: Partial<RestApiOptions>, modules: IRestApiModules) {
+    // Apply opts defaults
+    const opts = {
+      ...restApiOptionsDefault,
+      ...Object.fromEntries(Object.entries(optsArg).filter(([_, v]) => v != null)),
```

Comment:
since optsArg is a `Partial` type, any of its properties can be undefined. For example the port may not be set, and it would be undefined, which would then override the default port value in `restApiOptionsDefault` to be undefined.